metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1211,8 +1211,16 @@ class MetalsLanguageServer(
           scribe.error(s"error processing $sourceUri", e)
       }
     }
-    // Remove unused toplevel symbols from cache
-    tables.jarSymbols.deleteNotUsedTopLevels(usedJars.toArray)
+    // Schedule removal of unused toplevel symbols from cache
+    sh.schedule(
+      new Runnable {
+        override def run(): Unit = {
+          tables.jarSymbols.deleteNotUsedTopLevels(usedJars.toArray)
+        }
+      },
+      2,
+      TimeUnit.SECONDS
```

Comment:
minor, but you could use a SAM and this becomes:

```scala
sh.schedule(
  () => tables.jarSymbols.deleteNotUsedTopLevels(usedJars.toArray),
  2,
  TimeUnit.SECONDS
)
```

Reply:
Some overloads on `ExecutorService` don't work with SAM, I can't remember which exact one. I don't mind `new Runnable { ... }`